include/kokkos-utils/callbacks/TimerListenerBeginEnd.hpp

```diff
@@ -0,0 +1,74 @@
+#ifndef KOKKOS_UTILS_CALLBACKS_TIMERLISTENERBEGINEND_HPP
+#define KOKKOS_UTILS_CALLBACKS_TIMERLISTENERBEGINEND_HPP
+
+#include "kokkos-utils/callbacks/EventBeginEndEventIdMatcher.hpp"
+#include "kokkos-utils/callbacks/EventOnExecMatcher.hpp"
+#include "kokkos-utils/callbacks/TimerListener.hpp"
+#include "kokkos-utils/timer/Duration.hpp"
+#include "kokkos-utils/timer/Timer.hpp"
+
+namespace Kokkos::utils::callbacks
+{
+
+//! Timer for events that are enqueued on @ref exec.
+template <Kokkos::utils::concepts::ExecutionSpace Exec>
+struct EnqueuedEventTimer
+{
+    //! Start the timer. The event must be on @ref exec.
+    template <OnDeviceEvent EventType>
```

Comment:
We could also rename this concept `DeviceIdentifiableEvent`.

Reply:
If you make this change, the timer can also be renamed `DeviceIdentifiableEventTimer`.

Reply:
I guess it would be better to call it `EnqueuedEvent` ?

Reply:
OK. Made it `EnqueuedEvent`.